app/export.go

```diff
@@ -118,26 +177,70 @@ func (a *App) BulkExport(ctx request.CTX, writer io.Writer, outPath string, opts
 	if err != nil {
 		return err
 	}
-
+	exportFilename := a.GetID() + "_export.log"
+	var missingAttachments []LogFormat
 	if opts.IncludeAttachments {
-		ctx.Logger().Info("Bulk export: exporting file attachments")
+		mlog.Info("Bulk export: exporting file attachments")
 		for _, attachment := range attachments {
-			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
+			err := a.ExportAttachments(attachment, exportFilename, &missingAttachments, outPath, zipWr)
+			if err != nil {
 				return err
 			}
 		}
 		for _, attachment := range directAttachments {
-			if err := a.exportFile(outPath, *attachment.Path, zipWr); err != nil {
+			err := a.ExportAttachments(attachment, exportFilename, &missingAttachments, outPath, zipWr)
+			if err != nil {
 				return err
 			}
 		}
-		for _, emojiPath := range emojiPaths {
-			if err := a.exportFile(outPath, emojiPath, zipWr); err != nil {
+		for _, emoji := range emojiList {
+			isExist, err := a.FileExists(*emoji.Path)
+			if err != nil {
+				return err
+			}
+			if !isExist {
+				createdAtTime, err := strconv.ParseInt(strconv.FormatInt(emoji.CreateAt, 10), 10, 64)
+				if err != nil {
+					return model.NewAppError("parseTime", "Failed to parse createdAt time", nil, err.Error(), http.StatusInternalServerError)
+				}
+				updatedAtTime, err := strconv.ParseInt(strconv.FormatInt(emoji.UpdateAt, 10), 10, 64)
+				if err != nil {
+					return model.NewAppError("parseTime", "Failed to parse updatedAt time", nil, err.Error(), http.StatusInternalServerError)
+				}
+				data := LogFormat{
+					Id:        emoji.Id,
+					CreatorId: emoji.CreatorId,
+					CreateAt:  time.UnixMilli(createdAtTime),
+					UpdateAt:  time.UnixMilli(updatedAtTime),
+					Path:      *emoji.Path,
+					Name:      *emoji.Name,
+				}
+				missingAttachments = append(missingAttachments, data)
+				mlog.Warn(fmt.Sprintf("attachment %v not found and will not be exported", *emoji.Path))
+				continue
+			}
+			if err := a.exportFile(outPath, *emoji.Path, zipWr); err != nil {
 				return err
 			}
 		}
 	}
-
+	if len(missingAttachments) != 0 {
+		allMissingAttachments := MissingAttachments{
```

Comment:
dont create a variable used that directly

Reply:
Ok